@ajoubertza The latest pipeline was set to build with tango 9.2.5a in `.travis.yml`. Now the test fail with a link error:

```
E   ImportError: /home/travis/build/tango-controls/pytango/tango/_tango.so: undefined symbol: _ZN5Tango10DeviceDatarsERPKNS_18DevVarBooleanArrayE
```

which let me think that the CI either used a cached build of pytango to run the tests (scarry) or that the build and test environments are different. Any clue?

--- a/.travis.yml
+++ b/.travis.yml
@@ -7,7 +7,7 @@ branches:
 
 env:
   global:
-    - TANGO_VERSION=9.2.5a
+    - TANGO_VERSION=9.3.2alpha.1
   matrix:
     - PYTHON_VERSION=2.7
     - PYTHON_VERSION=3.5

--- a/.travis.yml
+++ b/.travis.yml
@@ -55,7 +55,7 @@ before_script:
   # Make sure old_ext exists
   - mkdir -p cached_ext
   # Touch the .so files if the extension hasn't changed
-  - diff cached_ext ext && find build -name _tango*.so -printf "touching %p\n" -exec touch {} + || true
+  - diff -r cached_ext ext && find build -name _tango*.so -printf "touching %p\n" -exec touch {} + || true
 
 script:
   # Build the package

--- a/ext/server/command.cpp
+++ b/ext/server/command.cpp
@@ -220,22 +220,16 @@ void extract_scalar<Tango::DEV_ENCODED>(const CORBA::Any &any, boost::python::ob
 
     if ((any >>= data) == false)
         throw_bad_type(Tango::CmdArgTypeName[Tango::DEV_ENCODED]);
-    
-    bopy::str encoded_format(data[0].encoded_format);
 
-#if PY_MAJOR_VERSION >= 3
+    boost::python::str encoded_format(data[0].encoded_format);
     boost::python::object encoded_data(
         boost::python::handle<>(
             PyBytes_FromStringAndSize(
-                (char*)data[0].encoded_data.get_buffer(),
+                (const char*)data[0].encoded_data.get_buffer(),
                 data[0].encoded_data.length()
             )
         )
     );
-#else
-    bopy::str encoded_data((const char*)data[0].encoded_data.get_buffer(),
-                           data[0].encoded_data.length());
-#endif
 
     o = boost::python::make_tuple(encoded_format, encoded_data);
 }

